grid-demo/src/app/angular/final-demo/final-demo.component.ts

```diff
@@ -0,0 +1,30 @@
+import { ListRange } from '@angular/cdk/collections';
+import { AfterViewInit, ChangeDetectorRef, Component } from '@angular/core';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+import { Vm, columns } from 'src/app/data/vm.generator';
+import { LazyLoadService } from './../lazy-loading-rows/lazy-load.service';
+
+@Component({
+  templateUrl: './final-demo.component.html',
+  styleUrls: ['./final-demo.component.scss'],
+  providers: [LazyLoadService],
+})
+export class AngularFinalDemoComponent {
+  readonly data: Observable<{ vms: Vm[]; totalResults: number; loadedCount: number }>;
+  columns = columns.map(column => ({ ...column, hidden: false }));
+  clrDgSelectable = true;
+
+  constructor(private readonly lazyLoadService: LazyLoadService, changeDetectorRef: ChangeDetectorRef) {
```

Comment:
`changeDetectorRef` should be marked `private readonly`.

Reply:
I don't need it to be a property though, it's just a constructor parameter

Reply:
Oh right, I am dumb. Don't mind me.